controllers/authorController.js

```diff
@@ -15,35 +19,196 @@ exports.author_list = function(req, res) {
 
 // Display detail page for a specific Author.
 exports.author_detail = function(req, res) {
-    res.send('NOT IMPLEMENTED: Author detail: ' + req.params.id);
+    
+  async.parallel({
+    author: function(callback) {
+        Author.findById(req.params.id)
+          .exec(callback)
+    },
+    authors_blogposts: function(callback) {
+      BlogPost.find({ 'author': req.params.id },'title summary')
+      .exec(callback)
+    },
+}, function(err, results) {
+    if (err) { return next(err); } // Error in API usage.
+    if (results.author==null) { // No results.
+        let err = new Error('Author not found');
+        err.status = 404;
+        return next(err);
+    }
+    // Successful, so render.
+    res.render('author_detail', { title: 'Author Detail', author: results.author, author_blogposts: results.authors_blogposts } );
+});
 };
 
 // Display Author create form on GET.
-exports.author_create_get = function(req, res) {
-    res.send('NOT IMPLEMENTED: Author create GET');
+exports.author_create_get = function(req, res, next) {       
+  res.render('author_form', { title: 'Create Author'});
 };
 
+
 // Handle Author create on POST.
-exports.author_create_post = function(req, res) {
-    res.send('NOT IMPLEMENTED: Author create POST');
-};
+exports.author_create_post = [
+
+  // Validate and sanitise fields.
+  body('first_name').trim().isLength({ min: 1 }).escape().withMessage('First name must be specified.')
+      .isAlphanumeric().withMessage('First name has non-alphanumeric characters.'),
+  body('family_name').trim().isLength({ min: 1 }).escape().withMessage('Family name must be specified.')
+      .isAlphanumeric().withMessage('Family name has non-alphanumeric characters.'),
+  body('date_of_birth', 'Invalid date of birth').optional({ checkFalsy: true }).isISO8601().toDate(),
+  body('date_of_death', 'Invalid date of death').optional({ checkFalsy: true }).isISO8601().toDate(),
```

Comment:
Agreed it's not relevant.  It was in the tutorial I used as a basis for this project and I thought it was good to include so that in my models I wasn't only working with strings.  So used for practice but would remove at some point before this goes live.